public/app/core/components/TimelineChart/utils.test.ts

```diff
@@ -45,6 +46,21 @@ describe('prepare timeline graph', () => {
     expect(info.warn).toEqual('Data does not have a time field');
   });
 
+  it('errors if more than 2 time fields', () => {
+    const frames = [
+      toDataFrame({
+        fields: [
+          { name: 'a', type: FieldType.time, values: [1, 2, 3] },
+          { name: 'b', type: FieldType.string, values: ['a', 'b', 'c'] },
+          { name: 'c', type: FieldType.time, values: [2, 4, 6] },
+          { name: 'd', type: FieldType.time, values: [3, 6, 9] },
+        ],
+      }),
+    ];
+    const info = prepareTimelineFields(frames, true, timeRange, theme);
+    expect(info.warn).toEqual('Data has too many time fields, expected at most two');
+  });
```

Comment:
I'm not sure if this is actually the right semantics. Can someone advise? If it isn't, could you share more in which situation having >2 time fields is valid?

Reply:
https://github.com/grafana/grafana/pull/84130 :smile:

Reply:
I've seen that PR before, but thanks for linking!

I think my confusion is about the case where there are >2 time fields, not when there are exactly 2 time fields (in a single frame).

Since I was previously modifying the `prepareTimelineFields` function, I thought it'd be nice to practice more defensive programming and make `prepareTimelineFields` explicitly handle the >2 case, hence the code modification and the test case I added here.

I have now reverted `prepareTimelineFields` to the original state, so I'm going to resolve this conversation.

I'll leave this comment here anyway just in case you think it's something worth revisiting at some point.